core/src/martian/multipart.cljc

```diff
@@ -0,0 +1,50 @@
+(ns martian.multipart
+  "An adaptor that prepares multipart content for all supported HTTP clients"
+  (:require [clojure.java.io :as io])
+  (:import (java.io File InputStream)))
+
+;; http-kit = {String | File, InputStream, byte[] | ByteBuffer! | ~Number!~}
+;; clj-http = {String | File, InputStream, byte[] | o.a.h.e.m.c.ContentBody}
+;; hato     = {String | File, InputStream, byte[] | URL, URI, Socket, Path!}
+;; bb/http  = {String | File, InputStream, byte[] | URL, URI, Socket, Path?}
+
```

Comment:
A conceptual diagram of the content types supported by all target HTTP clients. Strings, common binary types, extra binary types (everything that can be coerced to `InputStream`) and custom types (only for `clj-http` and `http-kit`).